src/Symfony/Component/Notifier/Transport/Dsn.php

```diff
@@ -96,6 +97,15 @@ public function getOption(string $key, $default = null)
         return $this->options[$key] ?? $default;
     }
 
+    public function getRequiredOption(string $key)
+    {
+        if (!isset($this->options[$key]) || !$this->options[$key]) {
```

Comment:
hmm.. `0` could be a valid value for a `timeout` option

Reply:
I will add testcases, thanks for the hint with the timeout, will add this an change the code parts accordingly 👍🏻